.dotnet/src/Utility/StreamingResult.cs

```diff
@@ -12,84 +11,28 @@ namespace OpenAI;
+        : base(response) 
     {
-        _rawResult = rawResult;
-        _asyncEnumerableSource = asyncEnumerableProcessor.Invoke(rawResult);
+        _asyncEnumerableSource = asyncEnumerableProcessor.Invoke(response);
     }
 
-    /// <summary>
-    /// Creates a new instance of <see cref="StreamingClientResult{T}"/> using the provided underlying HTTP response. The
-    /// provided function will be used to resolve the response into an asynchronous enumeration of streamed response
-    /// items.
-    /// </summary>
-    /// <param name="result">The HTTP response.</param>
-    /// <param name="asyncEnumerableProcessor">
-    /// The function that will resolve the provided response into an IAsyncEnumerable.
-    /// </param>
-    /// <returns>
-    /// A new instance of <see cref="StreamingClientResult{T}"/> that will be capable of asynchronous enumeration of
-    /// <typeparamref name="T"/> items from the HTTP response.
-    /// </returns>
-    internal static StreamingClientResult<T> CreateFromResponse(
-        ClientResult result,
-        Func<ClientResult, IAsyncEnumerable<T>> asyncEnumerableProcessor)
+    internal static StreamingEventResult<T> CreateFromResponse(
+        PipelineResponse response,
+        Func<PipelineResponse, IAsyncEnumerable<T>> asyncEnumerableProcessor)
     {
-        return new(result, asyncEnumerableProcessor);
+        return new(response, asyncEnumerableProcessor);
     }
 
-    /// <summary>
-    /// Gets the underlying <see cref="PipelineResponse"/> instance that this <see cref="StreamingClientResult{T}"/> may enumerate
-    /// over.
-    /// </summary>
-    /// <returns> The <see cref="PipelineResponse"/> instance attached to this <see cref="StreamingClientResult{T}"/>. </returns>
-    public PipelineResponse GetRawResponse() => _rawResult.GetRawResponse();
-
-    /// <summary>
-    /// Gets the asynchronously enumerable collection of distinct, streamable items in the response.
-    /// </summary>
-    /// <remarks>
-    /// <para> The return value of this method may be used with the "await foreach" statement. </para>
-    /// <para>
-    /// As <see cref="StreamingClientResult{T}"/> explicitly implements <see cref="IAsyncEnumerable{T}"/>, callers may
-    /// enumerate a <see cref="StreamingClientResult{T}"/> instance directly instead of calling this method.
-    /// </para>
-    /// </remarks>
-    /// <returns></returns>
-    public IAsyncEnumerable<T> EnumerateValues() => this;
-
-    /// <inheritdoc/>
-    public void Dispose()
-    {
-        Dispose(disposing: true);
-        GC.SuppressFinalize(this);
-    }
-
-    /// <inheritdoc/>
-    protected virtual void Dispose(bool disposing)
-    {
-        if (!_disposedValue)
-        {
-            if (disposing)
-            {
-                _rawResult?.GetRawResponse()?.Dispose();
-            }
-            _disposedValue = true;
-        }
-    }
+    // TODO: Handle disposal via Enumerator?  Validate that this will work.
```

Comment:
This is an open question we'll still need to resolve - whether or not to make StreamingClientResult<T> implement IDisposable.

Reply:
From the earlier-era Toub discussions, I think we'll end up needing `IDisposable` (or have a variant, equivalent mechanism, or whatever else). We looked at the option of implicitly using the end of enumeration as the sole disposal mechanism (and the existing implementation does opportunistically do it at the end of enumeration), but the ergonomics around making that an explicit lifetime requirement were unfavorably received. I think Krzysztof also mentioned that leaving an unbuffered response floating was a substantial issue.